docs/developer/CI.md

```diff
@@ -0,0 +1,410 @@
+please [contact here]()
+
+The following document assumes you have access to both the CASTEP source and the
+ANVIL CI system.
+
+!!!note
+
+    The ANVIL service is an STFC-cloud managed system and requests to register will
+    be considered by them on a case-by-case basis and are out of control of the CASTEP
+    team.
+
+## Personal Pipeline
+
+For the purposes of development testing you can request a personal pipeline in
+order to be able to rapidly test in the variety of environments under which the
+main CASTEP testing is performed.
+
+This is required for e.g. [modifying the `Jenkinsfile`](#contributing-to-the-jenkinsfile)
+
+Access to this pipeline will require an ANVIL account and a reasonable cause for
+the request.
+
+## BitBucket pipelines
+
+In addition to the main Jenkins tests, there is also a subset of short, fast
+tests which are run on BitBucket every commit to ascertain functionality
+quickly.
+
+The BitBucket pipelines are managed through the `bitbucket-pipelines.yml` in the
+root CASTEP folder.
+
+They currently run:
+
+- `lint` -- Run the [castep-linter](https://pypi.org/project/castep-linter/) on
+  the source.
+- `build-castep` -- A `debug` build of CASTEP with included LibXC
+- `build-castep-python` -- `debug` build of CASTEP's Python bindings
+- `test-castep` -- Run of CASTEP's `quick` test suite.
+- `test-python` -- Run of CASTEP's `python` test suite.
+
+All of these are performed on a single image of a Linux system compiled with
+`gfortran` and static `LAPACK` and `BLAS` libraries. They are not to be
+considered comprehensive.
+
+## Jenkins Pipelines
+
+The tests run in the course of CI are controlled through the `Jenkinsfile` in
+the root CASTEP folder.
+
+### Build keys
+
+The Jenkins tests are able to be run on a variety of different system
+configurations. In order to select a particular configuration, it is necessary
+to specify the key which identifies the configuration.
+
+The format of these keys is:
+
+```
+agent;arch;opt;comms#comment
+```
+
+Where `agent` is one of:
+
+- `rocky8` -- Rocky Linux 8
+- `rocky8-gpu` -- Rocky Linux 8 with GPU support
+- `rocky9` -- Rocky Linux 9
+- `icdpacewin` -- Windows
+
+and `arch`, `opt` and `comms` correspond to the CASTEP build system variables
+`ARCH`, `OPT` and `COMMS`. `comment` is used to distinguish otherwise similar
+variants and uniquely identify a particular configuration as well as apply
+modifications to the build (see [comment tags](#comment-tags)).
+
+It should be noted, however, that only certain combinations of variables are
+available on certain agents. For a full set of available combinations it is
+necessary to look into the `Jenkinsfile` itself as this list is frequently
+updated and changed.
+
+### Build Artefacts
+
+During a Jenkins run, various files will be preserved, making them available to
+download.
+
+Currently the artefacts that are preserved are:
+
+- Coverage reports
+- Linting reports
+- Failed test data
+- FORD Documentation
+
+### Reports
```

Comment:
A picture is worth 1000 words. Can we include a screenshot with some annotations to point out/explain the components including "doughnut" and the   stage view.

Actually, some more overall comments might as well go here. The document is currently missing some essentials.

1. The URL of the CASTEP Jenkins instance
2. Basic explanation of the web page view and definitions. What is the main entry view called? What do the colours mean?
3. Text and drop down Navigation links below the main Jenkins banner on the click-through page - best as an annotation on a page view screenshot,
4. Sidebar links - need a note (on the screenshot) that the contents are context-dependent.
5. Where to find Linter views.

And there really needs to be a basic "howto" on how to examine the logs to find out why a compile failed, find out which test(s) failed.  "What do I do if my PR fails the bitbucket pipeline" and "What do i do if my PR fails the Anvil pipeline?"